src/oscar/apps/checkout/views.py

```diff
@@ -194,7 +195,8 @@ def form_valid(self, form):
         return super(ShippingAddressView, self).form_valid(form)
 
 
-class UserAddressUpdateView(CheckoutSessionMixin, generic.UpdateView):
+class UserAddressUpdateView(
+        LoginRequiredMixin, CheckoutSessionMixin, generic.UpdateView):
```

Comment:
This is required for https://github.com/django-oscar/django-oscar/blob/master/tests/functional/checkout/customer_checkout_tests.py#L92 to pass. From my understanding updating an address is only used by a user with an account.